sql/plan/showvariables.go

```diff
@@ -110,3 +127,35 @@ func (sv *ShowVariables) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, er
 
 	return sql.RowsToRowIter(rows...), nil
 }
+
+func filterVariables(ctx *sql.Context, sessVars map[string]interface{}, filter sql.Expression) (map[string]interface{}, error) {
+	filteredMap := make(map[string]interface{})
+	for k, v := range sessVars {
+		updatedFilter, _, err := replaceUnresolvedColumnsWithKey(filter, k)
```

Comment:
If i am reading this correctly, we are representing `variable_name > 'variable'` as `expression.Equals(key, 'variable')`; i.e. we are constructing a new equals filter for every single key in the map. For the specific tests you have, it might make more sense to split this up into two parts: 1) during parsing, validate that there are no unresolved columns other than `variable_name` and replace it with a `expression.GetField(0)`; 2) call the in-place expression on a `sql.Row(k)`. The GetField will be replaced with the key value, we don't have to do semantic checking (make sure no UnresolvedColumns) every time, and we don't have to create a whole new filter every time. If the value is a possible input to a show variables WHERE expression, that would be ~GetField(0), but you don't test that so I'm assuming that isn't allowed.